keyboards/windx65/keymaps/thewerther/keymap.c

```diff
@@ -0,0 +1,183 @@
+                    register_code(KC_SPC);
+                    return false;
+                } else {
+                    return true;
+                }
+            } else { // key released
+                if ((get_mods() & MOD_BIT(KC_LSFT)) == MOD_BIT(KC_LSFT)) {
+                    if (shift_space_pressed) {
+                        unregister_code(KC_SPC);
+                        return true;
+                    }
+                } else {
+                    return true;
+                }
+            }
+            return true;
+
+        case KC_LSHIFT:
+            if (!record->event.pressed) {
+                if (shift_space_pressed) {
+                    unregister_code(KC_SPC);
+                    shift_space_pressed = false;
+                }
+            }
+            return true;
+        case BASE_LAYER_CHANGE:
+            if (record->event.pressed) {
+                if (get_highest_layer(default_layer_state) == _DEFAULT_LAYER) {
+                    set_single_persistent_default_layer(_NON_FUNC_LAYER);
+                }
+                else {
+                    set_single_persistent_default_layer(_DEFAULT_LAYER);
+                }
+                return false;
+            }
+        default:
+            return true;
+    }
+    return true;
+}
+
+void matrix_scan_user(void) {
+#   ifdef RGBLIGHT_ENABLE
+    // Set GRAVE key to red when _NON_FUNC_LAYER is active
+    if (get_highest_layer(default_layer_state) == _NON_FUNC_LAYER) {
+        rgblight_sethsv_noeeprom(HSV_RED);
+        rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+    }
+#   endif
+}
+
+uint16_t get_tapping_term(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case LT(_FUNC_LAYER, KC_SPC):
+            // When pressing shift and space set tapping term to 0 in order to
+            // be able to press space faster consecutively
+            if ((get_mods() & MOD_BIT(KC_LSFT)) == MOD_BIT(KC_LSFT)) {
+                return 0;
+            } else {
+                return TAPPING_TERM;
+            }
+        default:
+            return TAPPING_TERM;
+    }
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+#   ifdef RGBLIGHT_ENABLE
+    switch (get_highest_layer(state)) {
+        case _NON_FUNC_LAYER:
+            rgblight_sethsv_noeeprom(HSV_RED);
+            rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+            break;
+        case _FUNC_LAYER:
+            rgblight_sethsv_noeeprom(HSV_PURPLE);
+            rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+            break;
+        case _SETTINGS_LAYER:
+            rgblight_sethsv_noeeprom(HSV_WHITE);
+            rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+            break;
+        case _DEFAULT_LAYER:
+            rgblight_sethsv_noeeprom(HSV_MAGENTA);
+            rgblight_mode_noeeprom(RGBLIGHT_MODE_RAINBOW_MOOD);
+            break;
+    }
+#   endif
+    return state;
+}
+
```

Comment:
```suggestion

```
This should be in its own separate PR as a keymap PR if this board will be added.

Reply:
> submitters can have a personal (or bells-and-whistles) keymap showcasing capabilities in the same PR but it shouldn’t be embedded in the ‘default’ keymap

This is fine to include.